gpflow/experimental/check_shapes/parser.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2022 The GPflow Contributors. All Rights Reserved.
```

Comment:
I'm reluctant to add all this extra code to GPflow. Will make it harder to maintain for other people in the future. Why do we need this custom parser? What functionality would we lose by using the one in lark?